### DISCUSSION TITLE
Feature request: Add progress callback to FileSystem.uploadAsync

### DISCUSSION THREAD

**Discussion by hirbod** [expo/expo#20022] [Proposals and Ideas]
I am missing an upload progress for `FileSystem.uploadAsync`. The download function has a progress callback. Currently, I am not able to show a percentage progress to my users. I think its better instead of a undetermined progress spinner.

**Comment by andreilupe** [comment]
+1 this would be super useful

**Comment by hirbod** [comment]
I totally forgot to close this. FileSystem already has an upload progress callback. You need to use `createUploadTask`

**Reply by bryanprimus** [reply]
is there a snippet for this?

**Reply by itsdanreed** [reply]
Here's an example of uploading to a server with the progress callback:

```
const options = {
    fieldName: 'file',
    headers: { 
        'Accept': 'application/json',
         'Authorization': `Bearer ${authState.token}`                     
    },
    httpMethod: 'POST',                
    uploadType: FileSystem.FileSystemUploadType.MULTIPART,
};

const callback = ({ totalBytesSent, totalBytesExpectedToSend }) => {
    const progressPercentage = (totalBytesSent / (totalBytesExpectedToSend || 1)) * 100;
    setProgress(progressPercentage.toFixed(2));
};

const task = FileSystem.createUploadTask(url, uri, options, callback);
const response = await task.uploadAsync();
const data = JSON.parse(response.body);
```

I think it's fairly self-explanatory, but happy to help if you have any questions.
